server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,17 +370,55 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = telemetryTracer.startSpan(
+                SPAN_NAME_INCOMING_HTTP_SERVER_REQUEST,
+                buildSpanAttributes(httpRequest)
+            );
+            httpRequestSpanScope.addSpanAttribute(SPAN_ATTR_KEY_REQ_INBOUND_EX, httpRequest.getInboundException() != null);
+            HttpChannel traceableHttpChannel = new TraceableHttpChannel(httpChannel, httpRequestSpanScope);
+            handleIncomingRequest(httpRequest, traceableHttpChannel, httpRequest.getInboundException());
+        }
+    }
+
+    private Attributes buildSpanAttributes(HttpRequest httpRequest) {
+        Attributes attributes = Attributes.create()
+            .addAttribute(SPAN_ATTR_KEY_URI, httpRequest.uri())
```

Comment:
Please use  the ones from `SemanticAttributes` instead

Reply:
I was thinking of extracting this out to Constants but didn't like the idea later and reverted the changes. Let me take a look at SemanticAttributes and see how can we use it.